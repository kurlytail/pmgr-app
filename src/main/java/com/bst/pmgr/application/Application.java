package com.bst.pmgr.application;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.data.rest.RepositoryRestMvcAutoConfiguration;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.web.context.request.RequestContextListener;

@SpringBootApplication
@ComponentScan("com.bst")
@EntityScan({"com.bst.pmgr.entities", "com.bst.persistence.model"})
//@EnableAutoConfiguration(exclude = RepositoryRestMvcAutoConfiguration.class)
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
    
    @Bean
    public RequestContextListener requestContextListener() {
        return new RequestContextListener();
    }
}
