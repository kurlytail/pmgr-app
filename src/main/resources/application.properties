logging.level.org.apache.http=DEBUG
logging.level.org.springframework.web=DEBUG
logging.level.org.hibernate=INFO
logging.level.org.springframework.security=INFO

purge.cron.expression=0 0 5 * * ?

# Uncomment the keys to enable captcha
google.recaptcha.key=6LdXCWsUAAAAAOQLM0pi-6CU5IjCakpWEMmoTgHC
google.recaptcha.secret=6LdXCWsUAAAAAGGEBS1aLjoRLx9SS0fml7EyJyQw
google.recaptcha.url=https://www.google.com/recaptcha/api/siteverify

################### JavaMail Configuration ##########################
spring.mail.host=smtp.gmail.com
spring.mail.port=587
spring.mail.protocol=smtp
spring.mail.username=ankamath2000@gmail.com
spring.mail.password=Gth1Kmth
spring.mail.properties.mail.transport.protocol=smtp
spring.mail.properties.mail.smtp.auth=true
spring.mail.properties.mail.smtp.starttls.enable=true
spring.mail.properties.mail.smtp.timeout=8000

spring.data.rest.base-path=/api
spring.thymeleaf.cache=false

pmgr.app.version=@project.version@
pmgr.app.serverName=localhost

#spring.datasource.driverClassName=com.microsoft.sqlserver.jdbc.SQLServerDriver
#spring.datasource.url=jdbc:sqlserver://pmgr.database.windows.net:1433;database=pmgr
#spring.datasource.username=pmgradmin
#spring.datasource.password=tu64;Q;9yx-RS'<T
#spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.SQLServerDialect
#spring.jpa.show-sql=true

spring.jpa.database-platform=org.hibernate.dialect.SQLServerDialect
spring.jpa.hibernate.ddl-auto=update
spring.datasource.url=jdbc:mysql://pmgr.cmax6xpkmikd.us-east-2.rds.amazonaws.com:3306/pmgr
spring.datasource.username=pmgradmin
spring.datasource.password=tu64;Q;9yx-RS'<T

server.port=9000

#pmgr.app.configuration.security=disabled

