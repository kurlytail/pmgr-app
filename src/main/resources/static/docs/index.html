<!DOCTYPE html>
<html lang="en">
  <head>
    <!-- Required meta tags -->
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <!-- End Required meta tags -->
    <!-- Begin SEO tag -->
    <title> Getting Started | Looper - Bootstrap 4 Admin Theme </title>
    <meta property="og:title" content="Getting Started">
    <meta name="author" content="Beni Arisandi">
    <meta property="og:locale" content="en_US">
    <meta name="description" content="Responsive admin theme build on top of Bootstrap 4">
    <meta property="og:description" content="Responsive admin theme build on top of Bootstrap 4">
    <link rel="canonical" href="//uselooper.com">
    <meta property="og:url" content="//uselooper.com">
    <meta property="og:site_name" content="Looper - Bootstrap 4 Admin Theme">
    <script type="application/ld+json">
      {
        "name": "Looper - Bootstrap 4 Admin Theme",
        "description": "Responsive admin theme build on top of Bootstrap 4",
        "author":
        {
          "@type": "Person",
          "name": "Beni Arisandi"
        },
        "@type": "WebSite",
        "url": "",
        "headline": "Getting Started",
        "@context": "http://schema.org"
      }
    </script>
    <!-- End SEO tag -->
    <!-- Favicons -->
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="apple-touch-icon.png">
    <link rel="shortcut icon" href="favicon.ico">
    <meta name="theme-color" content="#070B0F">
    <!-- End Favicons -->
    <!-- BEGIN BASE STYLES -->
    <link rel="stylesheet" href="vendor/bootstrap/css/bootstrap.min.css">
    <link rel="stylesheet" href="vendor/open-iconic/css/open-iconic-bootstrap.min.css">
    <link rel="stylesheet" href="vendor/font-awesome/css/fontawesome-all.min.css">
    <!-- END BASE STYLES -->
    <!-- BEGIN PLUGINS STYLES -->
    <link rel="stylesheet" href="vendor/highlight-js/styles/ir_black.css">
    <!-- END PLUGINS STYLES -->
    <!-- BEGIN THEME STYLES -->
    <link rel="stylesheet" href="stylesheets/main.min.css">
    <!-- END THEME STYLES -->
  </head>
  <body data-spy="scroll" data-target="#nav-content" data-offset="76">
    <!-- .app -->
    <div class="app">
      <!-- .app-header -->
      <header class="app-header bg-black">
        <!-- .top-bar -->
        <div class="top-bar">
          <!-- .top-bar-brand -->
          <div class="top-bar-brand">
            <a href="index.html">
              <span class="tile tile-md tile-img mr-2">
                <img src="apple-touch-icon.png" width="36" alt="">
              </span>
              <strong>Documentation</strong>
            </a>
          </div>
          <!-- /.top-bar-brand -->
          <!-- .top-bar-list -->
          <div class="top-bar-list">
            <!-- .top-bar-item -->
            <div class="top-bar-item px-2 d-md-none d-lg-none d-xl-none">
              <!-- toggle menu -->
              <button class="hamburger hamburger-squeeze" type="button" data-toggle="aside" aria-label="Menu" aria-controls="navigation">
                <span class="hamburger-box">
                  <span class="hamburger-inner"></span>
                </span>
              </button>
              <!-- /toggle menu -->
            </div>
            <!-- /.top-bar-item -->
            <!-- .top-bar-item -->
            <!-- .top-bar-item -->
            <div class="top-bar-item top-bar-item-right pr-3 pr-lg-4">
              <!-- .nav -->
              <ul class="header-nav nav">
                <li class="nav-item">
                  <a class="nav-link" href="support.html">
                    <i class="fa fa-life-ring d-sm-none"></i>
                    <span class="d-none d-sm-inline">Support</span>
                  </a>
                </li>
                <li class="nav-item has-notified">
                  <a class="nav-link" href="whats-new.html">
                    <i class="fa fa-bell d-sm-none"></i>
                    <span class="d-none d-sm-inline">What's new</span>
                  </a>
                </li>
              </ul>
              <!-- /.nav -->
              <a href="https://themes.getbootstrap.com/product/looper-responsive-admin-template" class="btn btn-dark border-0">Download Now</a>
            </div>
            <!-- /.top-bar-item -->
          </div>
          <!-- /.top-bar-list -->
        </div>
        <!-- /.top-bar -->
      </header>
      <!-- /.app-header -->
      <!-- .app-aside -->
      <aside class="app-aside">
        <!-- .aside-content -->
        <div class="aside-content">
          <!-- .aside-menu -->
          <section class="aside-menu p-3">
            <!-- .stacked-menu -->
            <nav id="stacked-menu" class="stacked-menu">
              <!-- .menu -->
              <ul class="menu">
                <!-- .menu-item -->
                <li class="menu-item has-active">
                  <a href="index.html" class="menu-link">
                    <span class="menu-text">Getting Started</span>
                  </a>
                </li>
                <!-- /.menu-item -->
                <!-- .menu-item -->
                <li class="menu-item">
                  <a href="basic-layout.html" class="menu-link">
                    <span class="menu-text">Basic Layout</span>
                  </a>
                </li>
                <!-- /.menu-item -->
                <!-- .menu-item -->
                <li class="menu-item">
                  <a href="build-tools.html" class="menu-link">
                    <span class="menu-text">Build Tools</span>
                  </a>
                </li>
                <!-- /.menu-item -->
                <!-- .menu-item -->
                <li class="menu-item">
                  <a href="theming.html" class="menu-link">
                    <span class="menu-text">Theming</span>
                  </a>
                </li>
                <!-- /.menu-item -->
                <!-- .menu-header -->
                <li class="menu-header">Components </li>
                <!-- /.menu-header -->
                <!-- .menu-item -->
                <li class="menu-item">
                  <a href="components-bootstrap.html" class="menu-link">
                    <span class="menu-text">Bootstrap</span>
                  </a>
                </li>
                <!-- /.menu-item -->
                <!-- .menu-item -->
                <li class="menu-item">
                  <a href="components-looper.html" class="menu-link">
                    <span class="menu-text">Looper</span>
                  </a>
                </li>
                <!-- /.menu-item -->
                <!-- .menu-item -->
                <li class="menu-item">
                  <a href="components-plugins.html" class="menu-link">
                    <span class="menu-text">Plugins</span>
                  </a>
                </li>
                <!-- /.menu-item -->
              </ul>
              <!-- /.menu -->
            </nav>
            <!-- /.stacked-menu -->
          </section>
          <!-- /.aside-menu -->
        </div>
        <!-- /.aside-content -->
      </aside>
      <!-- /.app-aside -->
      <!-- .app-main -->
      <main class="app-main">
        <!-- .wrapper -->
        <div class="wrapper">
          <!-- .page -->
          <div class="page has-sidebar">
            <!-- .page-inner -->
            <div class="page-inner">
              <!-- .page-title-bar -->
              <header class="page-title-bar">
                <h1 id="overview" class="page-title"> Overview </h1>
                <p class="text-muted"> Everthing you need to know to get up and running with Looper. </p>
                <button class="btn btn-dark btn-floated d-none d-sm-block d-xl-none" type="button" data-toggle="sidebar">
                  <i class="fa fa-th-list"></i>
                </button>
              </header>
              <!-- /.page-title-bar -->
              <!-- .page-section -->
              <div class="page-section">
                <!-- .section-block -->
                <div class="section-block">
                  <h2 class="section-title"> Intro </h2>
                  <p> Hi there! Thank you for using Looper Theme! This theme build on top of latest Bootstrap 4 to help you speed up your awesome project. Rather than focusing on pixels, you can focus on application logic. This theme has been lovingly and
                    carefully crafted by Meâ€”
                    <a href="https://twitter.com/StilearningTwit" target="_blank">Beni Arisandi</a> with focus on simplicity, user experience, interactions and flows. Since this theme is based on Bootstrap, you can does everything that Bootstrap itself does. There are also a lot of custom and new components with
                    usage example on
                    <a href="//uselooper.com" target="_blank">Demo Pages</a>, and easy to use build tools. </p>
                  <p> Please familiarize your self with following: </p>
                  <ul>
                    <li>If you are not familiar with Bootstrap. Please get an overview of
                      <a href="http://getbootstrap.com/docs/4.1/getting-started/introduction/" target="_blank">Official Bootstrap Documentations</a> including markup, styles, components, features and everthing that bootstrap does. </li>
                    <li>Explore the
                      <a href="//uselooper.com/component-general.html" target="_blank">Looper Demo</a> page (especially on Interface section) to familiarize your self with existing components and plugins which you can incorporate into your designs. </li>
                    <li>Review the
                      <a href="//uselooper.com/component-general.html" target="_blank">Loper Pages</a> section, each pages provides reusable class and components. You can minimalize your custom styles by understanding the flows. </li>
                    <li>Considering about limitation; Since this theme based on Bootstrap please see
                      <a href="http://getbootstrap.com/docs/4.1/getting-started/browsers-devices/" target="_blank">Browsers and Devices</a> that are supported by Bootstrap. </li>
                  </ul>
                  <p> For more explanation about using Looper, theming and custom build, read through the docs. Please let me know about your pain-point with Looper and how we can make it better for you. </p>
                  <div class="alert alert-info"> For me, writing in english is 10x more harder that writing in code. I apologize for any typo both on Demo Pages and this Documentation. </div>
                </div>
                <!-- /.section-block -->
                <hr class="my-5">
                <!-- .section-block -->
                <div class="section-block">
                  <h1 id="content" class="page-title"> Contents </h1>
                  <p> Once downloaded, unzip the compressed folder. On top level directory structure of Looper Theme, you'll see something like this: </p> <pre><code class="markdown">looper-v1.0.1/
â”œâ”€â”€ dist/
â”œâ”€â”€ src/
â”œâ”€â”€ .babelrc
â”œâ”€â”€ .bowerrc
â”œâ”€â”€ .editorconfig
â”œâ”€â”€ .gitattributes
â”œâ”€â”€ .gitignore
â”œâ”€â”€ README.md
â”œâ”€â”€ bower.json
â”œâ”€â”€ gulpfile.js
â””â”€â”€ package.json</code></pre>
                  <h2 class="section-title"> Dist </h2>
                  <p> The <code>dist</code> directory is where all compiling files goes. The easiest way to using this theme is copying the <code>assets</code> folder on this directory to your existing project. This directory also contains the <code>docs</code>                    folder. See our
                    <a href="build-tools.html">Gulp tasks</a> for more information about how this directory was created. </p>
                  <h2 class="section-title"> Src </h2>
                  <p> The <code>src</code> directory is where all
                    <strong>Source Files</strong> are located, including demo files, data generator, snippets, assets, and docs. </p>
                </div>
                <!-- /.section-block -->
                <hr class="my-5">
                <!-- .section-block -->
                <div class="section-block">
                  <h1 id="quick-start" class="page-title"> Quick Start </h1>
                  <p> Be sure you have minimal requirement with the latest version of Looper template. The easiest way to using this theme is copying the <code>assets</code> folder on <code>dist</code> directory to your existing project. Then following steps
                    below: </p>
                  <h2 class="section-title"> HTML5 doctype </h2>
                  <p> Looper utilizing certain HTML and CSS elements that requires a applications. So make sure that you use it at your project documents. </p> <pre><code class="html">&lt;!doctype html&gt;
&lt;html lang="en"&gt;
  ...
&lt;/html&gt;</code></pre>
                  <h2 class="section-title"> Responsive meta tag </h2>
                  <p> Looper is mobile first design, so you must include the
                    <strong>responsive viewport meta tag</strong> to your <code>&lt;head&gt;</code> </p> <pre><code class="html">&lt;meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"&gt;</code></pre>
                  <h2 class="section-title">
                  Stylesheets </h2>
                  <p> All css files that you get are the result of compilation of SCSS files. We have divided it into small pieces which are easy to customized. </p>
                  <p> Below is required css that you need to include into your project: </p> <pre><code class="html">&lt;!-- BEGIN BASE STYLES --&gt;
&lt;link rel="stylesheet" href="assets/vendor/bootstrap/css/bootstrap.min.css"&gt;
&lt;!-- END BASE STYLES --&gt;

&lt;!-- BEGIN PLUGINS STYLES --&gt;
&lt;!-- plugins styles goes here --&gt;
&lt;!-- END PLUGINS STYLES --&gt;

&lt;!-- BEGIN THEME STYLES --&gt;
&lt;link rel="stylesheet" href="assets/stylesheets/main.min.css"&gt;
&lt;link rel="stylesheet" href="assets/stylesheets/custom.css"&gt;
&lt;!-- END THEME STYLES --&gt;

&lt;!-- BEGIN PAGE LEVEL STYLES --&gt;
&lt;!-- your styles for specific page goes here --&gt;
&lt;!-- END PAGE LEVEL STYLES --&gt;</code></pre>
                  <p> You must add the css files in the order above. Actually, Looper only required <code>bootstrap.min.css</code> and <code>main.min.css</code>. You can use <code>custom.scss</code> to write your own styles. if you doesn't need a custom styles,
                    feel free to remove <code>custom.css</code> from your markup up. </p>
                  <h2 class="section-title"> Javascript </h2>
                  <p> Below is required JS files that you need to add to your markup when using Looper. </p> <pre><code class="html">&lt;!-- BEGIN BASE JS --&gt;
&lt;script src="assets/vendor/jquery/jquery.min.js"&gt;&lt;/script&gt;
&lt;script src="assets/vendor/bootstrap/js/popper.min.js"&gt;&lt;/script&gt;
&lt;script src="assets/vendor/bootstrap/js/bootstrap.min.js"&gt;&lt;/script&gt;
&lt;!-- END BASE JS --&gt;

&lt;!-- BEGIN PLUGINS JS --&gt;
&lt;script src="assets/vendor/stacked-menu/stacked-menu.min.js"&gt;&lt;/script&gt;
&lt;!-- END PLUGINS JS --&gt;

&lt;!-- BEGIN THEME JS --&gt;
&lt;script src="assets/javascript/main.min.js"&gt;&lt;/script&gt;
&lt;!-- END THEME JS --&gt;

&lt;!-- BEGIN PAGE LEVEL JS --&gt;
&lt;!-- your js for specific page goes here --&gt;
&lt;!-- END PAGE LEVEL JS --&gt;</code></pre>
                  <p> Our <code>main.js</code> is writed in modular js. Looper also provide blank template that you can use to write your own scripts (just duplicate and rename it) or you can just use your own style. </p>
                  <h2 class="section-title"> Starter template </h2>
                  <p> Put it all together and your pages should look like this: </p> <pre><code class="html">&lt;!doctype html&gt;
&lt;html lang="en"&gt;
  &lt;head&gt;
    &lt;!-- Required meta tags --&gt;
    &lt;meta charset="utf-8"&gt;
    &lt;meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"&gt;

    &lt;!-- BEGIN BASE STYLES --&gt;
    &lt;link rel="stylesheet" href="assets/vendor/bootstrap/css/bootstrap.min.css"&gt;
    &lt;!-- END BASE STYLES --&gt;

    &lt;!-- BEGIN PLUGINS STYLES --&gt;
    &lt;!-- plugins styles goes here --&gt;
    &lt;!-- END PLUGINS STYLES --&gt;

    &lt;!-- BEGIN THEME STYLES --&gt;
    &lt;link rel="stylesheet" href="assets/stylesheets/main.min.css"&gt;
    &lt;link rel="stylesheet" href="assets/stylesheets/custom.css"&gt;
    &lt;!-- END THEME STYLES --&gt;

    &lt;!-- BEGIN PAGE LEVEL STYLES --&gt;
    &lt;!-- your styles for specific page goes here --&gt;
    &lt;!-- END PAGE LEVEL STYLES --&gt;

    &lt;title&gt;Hello, world!&lt;/title&gt;
  &lt;/head&gt;
  &lt;body&gt;
    &lt;h1&gt;Hello, world!&lt;/h1&gt;

    &lt;!-- jQuery first, then Popper.js, then Bootstrap JS --&gt;
    &lt;!-- BEGIN BASE JS --&gt;
    &lt;script src="assets/vendor/jquery/jquery.min.js"&gt;&lt;/script&gt;
    &lt;script src="assets/vendor/bootstrap/js/popper.min.js"&gt;&lt;/script&gt;
    &lt;script src="assets/vendor/bootstrap/js/bootstrap.min.js"&gt;&lt;/script&gt;
    &lt;!-- END BASE JS --&gt;

    &lt;!-- BEGIN PLUGINS JS --&gt;
    &lt;script src="assets/vendor/stacked-menu/stacked-menu.min.js"&gt;&lt;/script&gt;
    &lt;!-- END PLUGINS JS --&gt;

    &lt;!-- BEGIN THEME JS --&gt;
    &lt;script src="assets/javascript/main.min.js"&gt;&lt;/script&gt;
    &lt;!-- END THEME JS --&gt;

    &lt;!-- BEGIN PAGE LEVEL JS --&gt;
    &lt;!-- your js for specific page goes here --&gt;
    &lt;!-- END PAGE LEVEL JS --&gt;
  &lt;/body&gt;
&lt;/html&gt;</code></pre> </div>
                <!-- /.section-block -->
              </div>
              <!-- /.page-section -->
            </div>
            <!-- /.page-inner -->
            <!-- .page-sidebar -->
            <div class="page-sidebar page-sidebar-fixed">
              <nav id="nav-content" class="nav flex-column mt-4">
                <a href="#overview" class="nav-link smooth-scroll">Overview</a>
                <a href="#content" class="nav-link smooth-scroll">Content</a>
                <a href="#quick-start" class="nav-link smooth-scroll">Quick start</a>
              </nav>
            </div>
            <!-- /.page-sidebar -->
          </div>
          <!-- /.page -->
        </div>
        <!-- /.wrapper -->
      </main>
      <!-- /.app-main -->
    </div>
    <!-- /.app -->
    <!-- BEGIN BASE JS -->
    <script src="vendor/jquery/jquery.min.js"></script>
    <script src="vendor/bootstrap/js/popper.min.js"></script>
    <script src="vendor/bootstrap/js/bootstrap.min.js"></script>
    <!-- END BASE JS -->
    <!-- BEGIN PLUGINS JS -->
    <script src="vendor/highlight-js/highlight.pack.js"></script>
    <!-- END PLUGINS JS -->
    <!-- BEGIN THEME JS -->
    <script src="javascript/main.min.js"></script>
    <!-- END THEME JS -->
    <script>
      hljs.initHighlightingOnLoad();
    </script>
  </body>
</html>