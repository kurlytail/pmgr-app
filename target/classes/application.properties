logging.level.org.apache.http=DEBUG
logging.level.org.springframework.web=INFO
logging.level.org.hibernate=ERROR


purge.cron.expression=0 0 5 * * ?


# Uncomment the keys to enable captcha
google.recaptcha.key.site=6LfaHiITAAAAAAgZBHl4ZUZAYk5RlOYTr6m2N34X
google.recaptcha.key.secret=6LfaHiITAAAAANpDTA_Zflwib95IhDqg2SNRLt4U

################### JavaMail Configuration ##########################
support.email=ankamath2000@gmail.com
spring.mail.host=smtp.gmail.com
spring.mail.port=587
spring.mail.protocol=smtp
spring.mail.username=ankamath2000@gmail.com
spring.mail.password=Gth1Kmth
spring.mail.properties.mail.transport.protocol=smtp
spring.mail.properties.mail.smtp.auth=true
spring.mail.properties.mail.smtp.starttls.enable=true
spring.mail.properties.mail.smtp.timeout=8000