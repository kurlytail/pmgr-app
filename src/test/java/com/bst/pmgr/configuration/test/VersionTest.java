package com.bst.pmgr.configuration.test;

import static io.github.jsonSnapshot.SnapshotMatcher.expect;
import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static io.github.jsonSnapshot.SnapshotMatcher.validateSnapshots;

import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;
import org.junit.jupiter.api.DisplayName;
import org.junit.runner.RunWith;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
import org.springframework.boot.web.server.LocalServerPort;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
import org.springframework.test.web.servlet.setup.MockMvcBuilders;
import org.springframework.web.context.WebApplicationContext;

import com.bst.pmgr.application.Application;
import com.bst.utility.test.lib.SeleniumTest;

@RunWith(SpringJUnit4ClassRunner.class)
@SeleniumTest(driver = ChromeDriver.class)
@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT, classes = Application.class)
public class VersionTest {
	@Autowired
	private WebDriver driver;

	@LocalServerPort
	private int port;

	@Autowired
	private WebApplicationContext context;

	@Before
	public void setup() {
		MockMvcBuilders.webAppContextSetup(context).build();
	}

	public String url(String path) {
		return "http://localhost:" + port + path;
	}

	@BeforeClass
	public static void startSnapshot() {
		start();
	}

	@AfterClass
	public static void stopSnapshot() {
		validateSnapshots();
	}

	@Test
	@DisplayName("Should provide the correct version in the test")
	public void shouldReturnCorrectVersion() throws Exception {
		driver.get(url("/auth/signup"));
		expect(driver.findElement(By.id("applicationVersion")).getText()).toMatchSnapshot();
	}

}
