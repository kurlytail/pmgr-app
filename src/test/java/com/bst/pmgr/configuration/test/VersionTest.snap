com.bst.pmgr.configuration.test.VersionTest.shouldReturnCorrectVersion=[
  "Dummy version"
]